text/3453-f16-and-f128.md

```diff
@@ -0,0 +1,124 @@
+- Feature Name: `f16_and_f128`
+- Start Date: 2023-07-02
+- RFC PR: [rust-lang/rfcs#3453](https://github.com/rust-lang/rfcs/pull/3453)
+- Rust Issue: [rust-lang/rfcs#2629](https://github.com/rust-lang/rfcs/issues/2629)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes adding new floating point types `f16` and `f128` into the core language and standard library.
+
+# Motivation
+[motivation]: #motivation
+
+The IEEE 754 standard defines many binary floating point formats. The most common of these types are the binary32 and binary64 formats, available in Rust as `f32` and `f64`. However, other formats are useful in various other scenarios. The binary16 format is useful for situations where storage compactness is important and low precision is acceptable. The binary128 format is useful for situations where high precision is needed, such as scientific computing contexts.
+
+The proposal is to add `f16` and `f128` types in Rust to represent IEEE 754 binary16 and binary128 respectively. Having `f16` and `f128` types in the Rust language would make Rust an optimal environment for more advanced use cases. Unlike third-party crates, this enables the compiler to perform optimizations for hardware with native support for these types, allows defining literals for these types, and would provide one single canonical data type for these floats, making it easier to exchange data between libraries.
```

Comment:
I would add something here about what exactly the use cases are. Just something ike how f128 is useful for HPC and scientific computing, and how f16 is useful for GPU/SIMD/storage

Reply:
Does this sentence not accomplish this?

> The binary16 format is useful for situations where storage compactness is important and low precision is acceptable. The binary128 format is useful for situations where high precision is needed, such as scientific computing contexts.

Reply:
Oh whoops yeah, I missed that you mentioned exactly that for f128.

Maybe a specific real application for f16, mentioning AI is an easy way to get people on board. Lol

Reply:
AI often uses the brain float format, so I think mentioning it here might be unsuitable.